### ISSUE TITLE
Nano: metric wrapper `NanoMetric` generalization issue

### ISSUE THREAD

**Issue by TheaperDeng** [intel-analytics/BigDL#6390]
`NanoMetric` is currenly a pytorch-specific wrapper. some issues may make it hard to use

1. I think we don't need to check the signature to make sure the parameter name is `"pred"` and `"target"`?
https://github.com/intel-analytics/BigDL/blob/db9d6c80b347964177a902ded6cd17a13b8171f8/python/nano/src/bigdl/nano/pytorch/inference/optimizer.py#L874-L877

This makes users needs to write a wrapper on their original metric function just to make sure the signature could pass the check.

2. The implemetation does not support any kind of multiple input dataloader
https://github.com/intel-analytics/BigDL/blob/db9d6c80b347964177a902ded6cd17a13b8171f8/python/nano/src/bigdl/nano/utils/inference/pytorch/metrics.py#L32-L35
